### PR TITLE
[MB-9038] UpdateMTOShipment can review billable weight

### PR DESCRIPTION
**Pull request by Ronolibert**
## Description

Update UpdateMTOShipment support editing billable_weight_cap and billable_weight_justification

## Setup
Postman:
- Use updateMTOShipment endpoint
- Provide `billable_weight_cap` and `billable_weight_justification`
- Send request
Payload should include `billable_weight_cap` and `billable_weight_justification`

## Code Review Verification Steps
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-9038) for this change

### CODE REVIEW

**Review by YanZ777 — COMMENTED** [review]
Could we also update some of the devseed data to include data for these new fields?

**Review by donaldthai — COMMENTED** [review]
So the changes look fine to me. One thing I noticed was that the shipment can only be updated when it's in `SUBMITTED` status. I think when the TIO does try to update the shipment, the status will be `APPROVED`.

Here's the code I'm referring to: https://github.com/transcom/mymove/blob/5e8b44bab9814f462af202548c6346d043e8ea9f/pkg/services/mto_shipment/mto_shipment_updater.go#L189-L188

I think a quick change would be to:

- Add check for TIO role
- Add check for `Approved` status for the shipment

File: pkg/handlers/ghcapi/internal/payloads/model_to_payload.go [hunk 1]
```diff
@@ -279,28 +279,30 @@ func BackupContact(contacts models.BackupContacts) *ghcmessages.BackupContact {
...
-		CreatedAt:                strfmt.DateTime(mtoShipment.CreatedAt),
-		UpdatedAt:                strfmt.DateTime(mtoShipment.UpdatedAt),
-		ETag:                     etag.GenerateEtag(mtoShipment.UpdatedAt),
-		DeletedAt:                handlers.FmtDateTimePtr(mtoShipment.DeletedAt),
-		ApprovedDate:             handlers.FmtDateTimePtr(mtoShipment.ApprovedDate),
+		ID:                          strfmt.UUID(mtoShipment.ID.String()),
+		MoveTaskOrderID:             strfmt.UUID(mtoShipment.MoveTaskOrderID.String()),
+		ShipmentType:                ghcmessages.MTOShipmentType(mtoShipment.ShipmentType),
+		Status:                      ghcmessages.MTOShipmentStatus(mtoShipment.Status),
+		BillableWeightCap:           handlers.FmtPoundPtr(mtoShipment.BillableWeightCap),
```
[the remark this answers is not in the document]
Comment (Ronolibert) [reply pkg/handlers/ghcapi/internal/payloads/model_to_payload.go:286]: Added BillableWeightCap and Justification to the same seed data scenario other teams are using

**Comment by Ronolibert** [comment]
@donaldthai Added a check for TIOs attempting to update a shipment that is not approved

**Review by YanZ777 — APPROVED** [review]

File: pkg/services/mto_shipment/mto_shipment_updater.go [hunk 2]
```diff
@@ -179,8 +187,9 @@ func (e StaleIdentifierError) Error() string {
 
 //CheckIfMTOShipmentCanBeUpdated checks if a shipment should be updatable
 func (f *mtoShipmentUpdater) CheckIfMTOShipmentCanBeUpdated(appCtx appcontext.AppContext, mtoShipment *models.MTOShipment, session *auth.Session) (bool, error) {
-	if session.IsOfficeApp() && session.IsOfficeUser() && session.Roles.HasRole(roles.RoleTypeServicesCounselor) {
-		if mtoShipment.Status != models.MTOShipmentStatusSubmitted {
+	if session.IsOfficeApp() && session.IsOfficeUser() {
+		if (session.Roles.HasRole(roles.RoleTypeServicesCounselor) && mtoShipment.Status != models.MTOShipmentStatusSubmitted) ||
+			(session.Roles.HasRole(roles.RoleTypeTIO) && mtoShipment.Status != models.MTOShipmentStatusApproved) {
```
Comment (donaldthai) [inline pkg/services/mto_shipment/mto_shipment_updater.go:192]: Seems like this should be a `&&` operator: `session.Roles.HasRole(roles.RoleTypeServicesCounselor) && mtoShipment.Status != models.MTOShipmentStatusSubmitted) &&
			(session.Roles.HasRole(roles.RoleTypeTIO) && mtoShipment.Status != models.MTOShipmentStatusApproved`

I tested this with a multi-role user (services counselor and TIO) and I wasn't able to update a submitted shipment. Using `&&` did the trick in letting a user with both services counselor and TIO roles edit a shipment in `SUBMITTED` and `APPROVED` statuses.

Comment (Ronolibert) [reply pkg/services/mto_shipment/mto_shipment_updater.go]: Ahh good catch! Updated to reflect this case

**Pull request merged**